frontend/pkg/frontend/frontend.go

```diff
@@ -582,7 +517,7 @@ func (f *Frontend) ArmSubscriptionPut(writer http.ResponseWriter, request *http.
 
 	// Clean up resources if subscription is deleted.
 	if subscription.State == arm.SubscriptionStateDeleted {
-		if err := f.DeleteAllResources(ctx, subscriptionID); err != nil {
+		if err := f.DeleteAllResources(ctx, writer, request, subscriptionID); err != nil {
```

Comment:
I think we can pass a `nil` request here, right?  (Or just the request parameter from `DeleteAllResources`.) That will prevent `Location` and  `Azure-AsyncOperation` headers from being added to the subscription PUT response.
```suggestion
		if err := f.DeleteAllResources(ctx, writer, nil, subscriptionID); err != nil {
```